module/model-elasticsearch/src/source.ts

```diff
@@ -460,7 +460,7 @@ export class ElasticsearchModelSource extends ModelSource {
   async getByQuery<T extends ModelCore>(cls: Class<T>, query: ModelQuery<T> = {}, failOnMany = true): Promise<T> {
     const res = await this.getAllByQuery(cls, { limit: 2, ...query });
     if (!res || res.length < 1 || (failOnMany && res.length !== 1)) {
-      throw new AppError(`Invalid number of results for find by id: ${res ? res.length : res}`, 'data');
+      throw new AppError(`Invalid number of results for find by id: ${res ? res.length : res}`, 'notfound');
```

Comment:
I think you would need to check res.length to determine if it's a data problem or a notfound problem:
e.g. 
```typescript
throw new AppError(`Invalid number of results for find by id: ${res ? res.length : res}`, res && res.length > 1 ? 'data' :  'notfound');
```

Reply:
Updated PR to implement this logic across the 3 different models. Went with a more verbose approach that reduces the number of conditional checks (don't need to check res again twice when throwing error) and more clearly shows the two errors occuring, let me know if you'd prefer it to be handled with the same solution from your previous comment.

Reply:
Cool, looking better.  Given the commonality, it will probably be best to move this to a more common point for consistency (and future model implementations).  I would suggest adding in a `ModelUtil` class at `model/src/util/model.ts`  (you would need to create the file), and pull the common check into there, and then leverage that.  There will probably be more things like this that could and should be in the `ModelUtil`.

Reply:
Btw I'm happy to make the changes myself, if you would rather me finish up the request.

Reply:
Did not have time to implement this change, it's good to see the issue fixed though